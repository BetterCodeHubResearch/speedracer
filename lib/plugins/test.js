const path = require('path')

const { createDir } = require('../.internal/util')

export default (options) => (reports, file) => {
  reports.forEach(report => {
    const dirname = path.join(path.dirname(file), options.snapshotsDirname)
    const filename = path.join(dirname, `${report.metadata.id}.speedracer`)

    return loadReport(filename).then(
      oldReport => {
        process.exit(0)
      },
      () => createDir(dirname).then(
        () => writeFile(filename, race.report.serialize())
      )
    )
  })
}
