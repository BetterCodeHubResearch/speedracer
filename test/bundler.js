// Packages
import getStream from 'get-stream'
import test from 'ava'

// Ours
import createBundleStream from '../lib/bundler'

test('create a es6 bundle', async t => {
  const content = await getStream(
    createBundleStream('./test/fixtures/bundles/es6.js', { port: 3001 })
  )
  t.snapshot(content)
})

test('create a common js bundle', async t => {
  const content = await getStream(
    createBundleStream('./test/fixtures/bundles/cjs.js', { port: 3001 })
  )
  t.snapshot(content)
})

test('bundle runner client', async t => {
  const content = await getStream(
    createBundleStream('./test/fixtures/bundles/runner-client.js', { port: 3001 })
  )
  t.snapshot(content)
})

test('bundle a runner client with a custom port', async t => {
  const content = await getStream(
    createBundleStream('./test/fixtures/bundles/runner-client.js', { port: 1337 })
  )
  t.snapshot(content)
})
