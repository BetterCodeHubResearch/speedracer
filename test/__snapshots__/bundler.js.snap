// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bundle a runner client with a custom port 1`] = `
"(function () {
'use strict';

class RunnerClient {
  constructor() {
    this.port = 1337;
    this.ws = new WebSocket(\`ws://localhost:\${this.port}\`);
    this.router = {
      'run:start:ack': () => this.startRun(),
      'run:end:ack': () => this.runNext()
    };
    this.runs = [];

    this.ws.onopen = () => {
      this.ws.onmessage = ({ data }) => this.onMessage(JSON.parse(data));
      this.send('file:start');
      this.runNext();
    };
  }

  enqueueRun(title, fn) {
    this.runs.push({ title, fn });
  }

  runNext() {
    this.currentRun = this.runs.shift();

    if (this.currentRun) {
      this.send('run:start', this.currentRun);
    }
    else {
      this.send('file:end');
      this.ws.close();
    }
  }

  startRun() {
    try {
      const res = this.currentRun.fn();
      if (res && res.then) {
        res.then(() => this.endRun());
      }
      else {
        this.endRun();
      }
    }
    catch (err) {
      this.runError(err);
    }
  }

  endRun() {
    this.send('run:end', this.currentRun);
  }

  runError(err) {
    this.send('run:error', this.currentRun, err);
    this.endRun();
  }

  onMessage(payload) {
    const { type } = payload;
    const handler = this.router[type];
    if (handler) {
      handler();
    }
  }

  send(type, run, err) {
    const payload = { type };
    if (run) payload.run = run;
    if (err) payload.err = err;
    this.ws.send(JSON.stringify(payload));
  }
}

const runner = new RunnerClient();

var runnerClient = (title, fn) =>
runner.enqueueRun(title, fn);

// For test purposes
var RunnerClient_1 = RunnerClient;

runnerClient.RunnerClient = RunnerClient_1;

runnerClient.end();

}());
"
`;

exports[`bundle runner client 1`] = `
"(function () {
'use strict';

class RunnerClient {
  constructor() {
    this.port = 3001;
    this.ws = new WebSocket(\`ws://localhost:\${this.port}\`);
    this.router = {
      'run:start:ack': () => this.startRun(),
      'run:end:ack': () => this.runNext()
    };
    this.runs = [];

    this.ws.onopen = () => {
      this.ws.onmessage = ({ data }) => this.onMessage(JSON.parse(data));
      this.send('file:start');
      this.runNext();
    };
  }

  enqueueRun(title, fn) {
    this.runs.push({ title, fn });
  }

  runNext() {
    this.currentRun = this.runs.shift();

    if (this.currentRun) {
      this.send('run:start', this.currentRun);
    }
    else {
      this.send('file:end');
      this.ws.close();
    }
  }

  startRun() {
    try {
      const res = this.currentRun.fn();
      if (res && res.then) {
        res.then(() => this.endRun());
      }
      else {
        this.endRun();
      }
    }
    catch (err) {
      this.runError(err);
    }
  }

  endRun() {
    this.send('run:end', this.currentRun);
  }

  runError(err) {
    this.send('run:error', this.currentRun, err);
    this.endRun();
  }

  onMessage(payload) {
    const { type } = payload;
    const handler = this.router[type];
    if (handler) {
      handler();
    }
  }

  send(type, run, err) {
    const payload = { type };
    if (run) payload.run = run;
    if (err) payload.err = err;
    this.ws.send(JSON.stringify(payload));
  }
}

const runner = new RunnerClient();

var runnerClient = (title, fn) =>
runner.enqueueRun(title, fn);

// For test purposes
var RunnerClient_1 = RunnerClient;

runnerClient.RunnerClient = RunnerClient_1;

runnerClient.end();

}());
"
`;

exports[`create a common js bundle 1`] = `
"var run = (function () {
'use strict';

var dep = () => {};


var dep$1 = Object.freeze({
	default: dep
});

var dep$2 = ( dep$1 && dep ) || dep$1;

dep$2();

var cjs = {

};

return cjs;

}());
"
`;

exports[`create a es6 bundle 1`] = `
"(function () {
'use strict';

var dep = () => {};

dep();

}());
"
`;
